Sources/Moya/Plugins/NetworkLoggerPlugin.swift

```diff
@@ -18,17 +18,21 @@ public final class NetworkLoggerPlugin: PluginType {
     public let cURL: Bool
 
     /// Initializes a NetworkLoggerPlugin.
-    public init(verbose: Bool = false, cURL: Bool = false, output: ((_ separator: String, _ terminator: String, _ items: Any...) -> Void)? = nil, requestDataFormatter: ((Data) -> (String))? = nil, responseDataFormatter: ((Data) -> (Data))? = nil) {
+    public init(verbose: Bool = false,
+                cURL: Bool = false,
+                output: NetworkLoggerOutput = DefaultNetworkLoggerOutput(),
+                requestDataFormatter: ((Data) -> (String))? = nil,
+                responseDataFormatter: ((Data) -> (Data))? = nil) {
```

Comment:
If we made it a `NetworkLoggerFormatter` we could throw these 2 properties in there as well to remain consistent

Reply:
Possibly?
`public init(verbose: Bool = false, cURL: Bool = false, formatter: NetworkLoggerFormatter = .defaultFormatter)`

Reply:
I like the idea for the generic `Formatter` class, but imo all of the properties should have a default value, as I don't usually want to override everything. Also, how do you feel about namespacing the `Formatter` object in the `NetworkLoggerPlugin`?
```swift
public final class NetworkLoggerPlugin: PluginType {
    class Formatter {}
}
```

Reply:
I'm fine with namespacing it, but then it can't be `open`. All the properties would have default properties, declared as vars so you can assign a different closure to them.